brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -100,43 +100,51 @@ public BrageEntryEventConsumer() {
+                                           ? attemptToUpdateExistingPublication(publicationRepresentation, s3Event)
+                                           : createNewPublication(publicationRepresentation, s3Event))
                    .orElse(fail -> handleSavingError(fail, s3Event));
     }
 
+    @SuppressWarnings("PMD.NullAssignment")
+    private void resetRuntime() {
+        brageRecordFile = null;
+        source = null;
+        publicationsToMerge = null;
+    }
+
     private static Publication getOnlyElement(List<Publication> publications) {
         return attempt(() -> Iterables.getOnlyElement(publications))
                    .orElseThrow(fail -> new DuplicatePublicationException(
                        DUPLICATE_PUBLICATIONS_MESSAGE,
                        fail.getException()));
     }
 
-    private boolean shouldMergePublications(Publication publication) throws NotFoundException {
-        var cristinIdentifier = getCristinIdentifier(publication);
+    private boolean shouldMergePublications(PublicationRepresentation publication) throws NotFoundException {
+        var cristinIdentifier = getCristinIdentifier(publication.nvaPublication());
         if (nonNull(cristinIdentifier)) {
             publicationsToMerge = resourceService.getPublicationsByCristinIdentifier(cristinIdentifier).stream()
-                                      .filter(item -> PublicationComparator.publicationsMatch(item, publication))
+                                      .filter(item -> PublicationComparator.publicationsMatch(item, publication.nvaPublication()))
                                       .toList();
             boolean shouldMerge = !publicationsToMerge.isEmpty();
             source = shouldMerge ? MergeSource.CRISTIN : MergeSource.NOT_RELEVANT;
             return shouldMerge;
         }
-        if (hasDoi(publication)) {
-            var shouldMerge= existingPublicationHasSameDoi(publication);
+        if (hasDoi(publication.nvaPublication())) {
+            var shouldMerge= existingPublicationHasSameDoi(publication.nvaPublication());
             source = shouldMerge ? MergeSource.DOI : MergeSource.NOT_RELEVANT;
             return shouldMerge;
         }
-        if (hasIsbn(publication)) {
-            var shouldMerge = existingPublicationHasSameIsbn(publication);
+        if (hasIsbn(publication.nvaPublication())) {
+            var shouldMerge = existingPublicationHasSameIsbn(publication.nvaPublication());
             source = shouldMerge ? MergeSource.ISBN : MergeSource.NOT_RELEVANT;
             return shouldMerge;
         } else {
-            var shouldMerge = existingPublicationHasSamePublicationContent(publication);
+            var shouldMerge = existingPublicationHasSamePublicationContent(publication.nvaPublication());
             source = shouldMerge ? MergeSource.SEARCH : MergeSource.NOT_RELEVANT;
             return shouldMerge;
         }
```

Comment:
I would rename publication -> publicationRepresentation. So we do publicationRepresentation.publication(), instead of publication.nvaPublication()